From 9040c05b52f1300c18ab51d91fd8224d1f2cb6e9 Mon Sep 17 00:00:00 2001
From: Ron Eldor <Ron.Eldor@arm.com>
Date: Tue, 15 Jan 2019 18:54:03 +0200
Subject: [PATCH 33/43] Force IPv4 in gnutls_cli srtp tests

Force using IPv4 in the GNU_CLI SRTP tests, as introduced for
other tests in #1918.
---
 tests/ssl-opt.sh | 71 ++++++++++++++++++++----------------------------
 1 file changed, 29 insertions(+), 42 deletions(-)

diff --git a/tests/ssl-opt.sh b/tests/ssl-opt.sh
index 6c49944a0..52d3306fe 100755
--- a/tests/ssl-opt.sh
+++ b/tests/ssl-opt.sh
@@ -7160,14 +7160,11 @@ run_test  "DTLS-SRTP all profiles supported. server doesn't support mki. openssl
           -C "dumping 'received mki' (8 bytes)" \
           -C "error"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP all profiles supported. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile" \
@@ -7175,14 +7172,11 @@ run_test  "DTLS-SRTP all profiles supported. gnutls client." \
           -s "server hello, adding use_srtp extension" \
           -c "SRTP profile: SRTP_AES128_CM_HMAC_SHA1_80"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports all profiles. Client supports all profiles, in different order. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_NULL_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_80:SRTP_NULL_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_NULL_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_80:SRTP_NULL_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile" \
@@ -7190,14 +7184,11 @@ run_test  "DTLS-SRTP server supports all profiles. Client supports all profiles,
           -s "server hello, adding use_srtp extension" \
           -c "SRTP profile: SRTP_NULL_HMAC_SHA1_80"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports all profiles. Client supports one profile. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile: MBEDTLS_SRTP_AES128_CM_HMAC_SHA1_32" \
@@ -7205,14 +7196,11 @@ run_test  "DTLS-SRTP server supports all profiles. Client supports one profile.
           -s "server hello, adding use_srtp extension" \
           -c "SRTP profile: SRTP_AES128_CM_HMAC_SHA1_32"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports one profile. Client supports all profiles. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=4 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=4 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile" \
@@ -7220,14 +7208,11 @@ run_test  "DTLS-SRTP server supports one profile. Client supports all profiles.
           -s "server hello, adding use_srtp extension" \
           -c "SRTP profile: SRTP_NULL_SHA1_32"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server and Client support only one matching profile. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=2 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=2 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile" \
@@ -7235,14 +7220,11 @@ run_test  "DTLS-SRTP server and Client support only one matching profile. gnutls
           -s "server hello, adding use_srtp extension" \
           -c "SRTP profile: SRTP_AES128_CM_HMAC_SHA1_32"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server and Client support only one different profile. gnutls client." \
-          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=1 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32" \
+          "$P_SRV dtls=1 use_srtp=1 srtp_force_profile=1 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -s "found srtp profile" \
@@ -7250,20 +7232,18 @@ run_test  "DTLS-SRTP server and Client support only one different profile. gnutl
           -S "server hello, adding use_srtp extension" \
           -C "SRTP profile:"
 
-# gnutls-cli resolves localhost as an IPv6 address, when enabled,
-# and fails to send messagges over UDP, causing DTLS negotiation to fail.
-# Force server to bind to IPv6 address
-requires_ipv6
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server doesn't support use_srtp extension. gnutls client" \
-          "$P_SRV dtls=1 debug_level=3 server_addr=::1" \
-          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
+          "$P_SRV dtls=1 debug_level=3" \
+          "$G_CLI -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32 --insecure 127.0.0.1" \
           0 \
           -s "found use_srtp extension" \
           -S "server hello, adding use_srtp extension" \
           -C "SRTP profile:"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP all profiles supported. gnutls server" \
           "$G_SRV -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 debug_level=3" \
@@ -7275,6 +7255,7 @@ run_test  "DTLS-SRTP all profiles supported. gnutls server" \
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports all profiles. Client supports all profiles, in different order. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_NULL_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_80:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 debug_level=3" \
@@ -7286,6 +7267,7 @@ run_test  "DTLS-SRTP server supports all profiles. Client supports all profiles,
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports all profiles. Client supports one profile. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_NULL_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_AES128_CM_HMAC_SHA1_80:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 srtp_force_profile=2 debug_level=3" \
@@ -7297,6 +7279,7 @@ run_test  "DTLS-SRTP server supports all profiles. Client supports one profile.
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server supports one profile. Client supports all profiles. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_NULL_HMAC_SHA1_80" \
           "$P_CLI dtls=1 use_srtp=30 debug_level=3" \
@@ -7308,6 +7291,7 @@ run_test  "DTLS-SRTP server supports one profile. Client supports all profiles.
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server and Client support only one matching profile. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 srtp_force_profile=2 debug_level=3" \
@@ -7319,6 +7303,7 @@ run_test  "DTLS-SRTP server and Client support only one matching profile. gnutls
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server and Client support only one different profile. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 srtp_force_profile=4 debug_level=3" \
@@ -7330,6 +7315,7 @@ run_test  "DTLS-SRTP server and Client support only one different profile. gnutl
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP server doesn't support use_srtp extension. gnutls server" \
           "$G_SRV -u" \
           "$P_CLI dtls=1 use_srtp=1 debug_level=3" \
@@ -7341,6 +7327,7 @@ run_test  "DTLS-SRTP server doesn't support use_srtp extension. gnutls server" \
           -C "error"
 
 requires_config_enabled MBEDTLS_SSL_DTLS_SRTP
+requires_gnutls
 run_test  "DTLS-SRTP all profiles supported. mki used. gnutls server." \
           "$G_SRV -u --srtp-profiles=SRTP_AES128_CM_HMAC_SHA1_80:SRTP_AES128_CM_HMAC_SHA1_32:SRTP_NULL_HMAC_SHA1_80:SRTP_NULL_SHA1_32" \
           "$P_CLI dtls=1 use_srtp=1 mki=542310ab34290481 debug_level=3" \
-- 
2.17.1

